xbmc/cores/paplayer/AudioDecoder.cpp

```diff
@@ -102,9 +102,30 @@ bool CAudioDecoder::Create(const CFileItem &file, int64_t seekOffset)
   /* allocate the pcmBuffer for 2 seconds of audio */
   m_pcmBuffer.Create(2 * blockSize * m_codec->m_SampleRate);
 
-  // set total time from the given tag
-  if (file.HasMusicInfoTag() && file.GetMusicInfoTag()->GetDuration())
-    m_codec->SetTotalTime(file.GetMusicInfoTag()->GetDuration());
+  if (file.HasMusicInfoTag())
+  {
+    // set total time from the given tag
+    if (file.GetMusicInfoTag()->GetDuration())
+      m_codec->SetTotalTime(file.GetMusicInfoTag()->GetDuration());
+
+    // update ReplayGain from the given tag if it's better then original (cuesheet)
+    ReplayGain rgInfo = m_codec->m_tag.GetReplayGain();
```

Comment:
1. We can't use const reference here. This is not read only.
2. MusicInfoTag.h contains include ReplayGain.h and we can't use predeclaration.